packages/flutter/lib/src/painting/text_scaler.dart

```diff
@@ -140,13 +140,17 @@ final class _ClampedTextScaler implements TextScaler {
     if (identical(this, other)) {
       return true;
     }
+    assert(minScale < maxScale);
```

Comment:
If this is the only place to enforce this constraint, we should do it where these two values are set instead. a crash here won't help us much in tracing down the bug

Reply:
This is enforced in the (private) constructor (if `minScale == maxScale` the public API will return a different type of `TextScaler`). The assert is to indicate this is a valid assumption one can make for readers. Does it make more sense to make this a code comment then?

Reply:
as long as there is not the first fail safe point, I am fine either way.